crates/hir/src/lib.rs

```diff
@@ -1087,6 +1088,27 @@ impl Field {
         Type::new(db, var_id, ty)
     }
 
+    pub fn ty_with_args(&self, db: &dyn HirDatabase, generics: impl Iterator<Item = Type>) -> Type {
```

Comment:
I'm not sure I like this as an API, the way it deals with const args is really weird :sweat_smile: and having it duplicated on every thing that has a type is also not great.

Reply:
That is true, however I couldn't really come up with something better right away, especially for stuff that get generics also from impl block.
```rust
impl<T> for Foo<T> {
    fn foo<U>(...) -> Bar<T, U> { ... }
}
```
I guess an option is to turn that impl block into impl block for concrete type, then do the same for function and then get the return value of the function normally. However that kind of duplicates all the definitions themselves so I don't really want to do it